src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -275,4 +280,21 @@ AbstractTableGenerator::SortOrderByTable AbstractTableGenerator::_sort_order_by_
 void AbstractTableGenerator::_add_constraints(
     std::unordered_map<std::string, BenchmarkTableInfo>& table_info_by_name) const {}
 
+bool AbstractTableGenerator::_table_is_sorted_by(const std::shared_ptr<Table>& table,
+                                                 const SortColumnDefinition& sort_definition) {
+  bool table_sorted = true;
+  for (ChunkID chunk_id{0}; chunk_id < table->chunk_count(); ++chunk_id) {
+    bool chunk_sorted = false;
+    for (const auto& sort_column_definition : table->get_chunk(chunk_id)->sorted_by()) {
+      if (sort_column_definition.column == sort_definition.column) {
+        Assert(sort_column_definition.sort_mode == sort_definition.sort_mode,
+               "Column is already sorted by another SortMode");
+        chunk_sorted = true;
+      }
+    }
+    table_sorted &= chunk_sorted;
```

Comment:
I am fine with this implementation as it is easy to understand. It could be made more efficient by returning `false` as early as the first chunk is not sorted.

Reply:
That was actually not what I meant but it's another optimization. I meant if `chunk_sorted` in 297 turns `table_sorted` to `false`, all further chunks will still be evaluated while `table_sorted` cannot turn true anymore. But you don't have to change the current implementation, it's your call.

Reply:
OK, got it :D